src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/ErrorProbabilities.java

```diff
@@ -3,36 +3,95 @@
+        // combine allele-wise
+        combinedErrorProbabilitiesByAllele = transpose(probabilitiesByTypeAndAllele.values().stream().collect(toList()))
+                .stream().map(
+                        alleleProbabilities -> alleleProbabilities.stream().map(p -> 1.0 - p).reduce(1.0, (a, b) -> a * b)).collect(Collectors.toList());
+        combinedErrorProbabilitiesByAllele.replaceAll(trueProb -> Mutect2FilteringEngine.roundFinitePrecisionErrors(1.0 - trueProb));
+    }
+
+    public List<Double> getCombinedErrorProbabilities() { return combinedErrorProbabilitiesByAllele; }
+    public List<Double> getTechnicalArtifactProbabilities() { return probabilitiesByTypeAndAllele.get(ErrorType.ARTIFACT); }
+    public List<Double> getNonSomaticProbabilities() { return probabilitiesByTypeAndAllele.get(ErrorType.NON_SOMATIC); }
+    public Map<Mutect2Filter, List<Double>> getProbabilitiesByFilter() { return alleleProbabilitiesByFilter; }
+
+    // helper functions for the few operations that still differ depending on whether the filter
+    // is per variant or allele
+    public Map<Mutect2Filter, List<Double>> getProbabilitiesForAlleleFilters() {
+        return getPartitionedProbabilitiesByFilter(false);
+    }
 
-        errorProbability = Mutect2FilteringEngine.roundFinitePrecisionErrors(1 - trueProbability);
+    public Map<Mutect2Filter, Double> getProbabilitiesForVariantFilters() {
+        return getPartitionedProbabilitiesByFilter(true).entrySet().stream()
+                .filter(entry -> entry.getValue() != null && !entry.getValue().isEmpty())
+                .collect(toMap(entry -> entry.getKey(), entry -> entry.getValue().get(0)));
     }
 
-    public double getErrorProbability() { return errorProbability; }
-    public double getTechnicalArtifactProbability() { return probabilitiesByType.get(ErrorType.ARTIFACT); }
-    public double getNonSomaticProbability() { return probabilitiesByType.get(ErrorType.NON_SOMATIC); }
-    public Map<Mutect2VariantFilter, Double> getProbabilitiesByFilter() { return probabilitiesByFilter; }
+    private Map<Mutect2Filter, List<Double>> getPartitionedProbabilitiesByFilter(boolean variantOnly) {
+        Map<Boolean, LinkedHashMap<Mutect2Filter, List<Double>>> groups =
+                alleleProbabilitiesByFilter.entrySet().stream().collect(Collectors.partitioningBy(
+                        entry -> Mutect2VariantFilter.class.isAssignableFrom(entry.getKey().getClass()),
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (a,b) -> a, LinkedHashMap::new)));
+        return groups.get(variantOnly);
+    }
 
+    // TODO would this be useful in a util class somewhere?
+    private static <T> List<List<T>> transpose(List<List<T>> list) {
+        // all lists need to be the same size
+        final int N = list.stream().mapToInt(l -> l.size()).max().orElse(-1);
+        if (list.stream().anyMatch(l -> l.size() != N)) {
```

Comment:
Should there be an error thrown inside this block?

Reply:
yes, is there an example you can refer me to for the best way to throw this error?

Reply:
I think something like
```
Utils.validateArg(list.stream().mapToInt(List::size).distinct().count() == 1, "lists are not the same size");
```
This throws an `IllegalArgumentException`, which seems like the right choice to me.